src/units/attack_type.cpp

```diff
@@ -99,6 +101,17 @@ attack_type::attack_type(const config& cfg) :
 		else
 			icon_ = "attacks/blank-attack.png";
 	}
+	if(cfg.has_attribute("alignment") && (cfg["alignment"] == "neutral" || cfg["alignment"] == "lawful" || cfg["alignment"] == "chaotic" || cfg["alignment"] == "liminal")){
+		alignment_str_ = cfg["alignment"].str();
+	} else if(self_){
```

Comment:
What is this check for? Why set weapon alignment based on `self_`?

Reply:
if self_ exist in this function(lua check) then what not alignment attribute valid, then unit alignment can be returned and second check will use an directly alignment, else alignment is empty and second check should be used in alignment() but i see your point if attribute empty or not valid it's unuse what alignment() make a complete chek and must return unit alignment in case of empty value, i will correct that.

Reply:
aalignment() can't have self_ in lua ccheck then i must doing that in init() for what attack alignment function perfect condition and what unit alignment remain value by default.